controllers/payment.controllers.js

```diff
@@ -99,7 +103,36 @@ async function newSetupIntent(req, res, next) {
 // payment from customer - one payment session - "donate" button
 async function newPaymentIntent(req, res, next) {
   const { stripeUserId, email } = req.user;
-  const { amount, description, paymentMethod } = req.body;
+  const { amount, description, last4 } = req.body;
+
+  let paymentMethods;
+  try {
+    paymentMethods = await stripe.customers.listPaymentMethods(stripeUserId, {
+      type: 'card',
+    });
+  } catch (err) {
+    console.log(err);
+    const error = new HttpError(
+      'Could not find your card. Please try again later',
+      500,
+    );
+    return next(error);
+  }
+
+  let paymentMethod;
+  for (let i = 0; i < paymentMethods.data.length; i++) {
+    if (paymentMethods.data[i].card.last4 == last4) {
+      paymentMethod = paymentMethods.data[i].id;
+    }
+  }
```

Comment:
fist of all: try to avoid using "==", use "===". 
The second one: try to avoid for loops, use array methods instead

Try this one, it should do the same, but check carefully

`const paymentMethod = paymentMethods.data.find(method => method.card.last4 === last4)?.id`

Reply:
Thanks, it's must be "experienced programmers power" because everything is working